packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -907,6 +907,11 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
   Element get renderViewElement => _renderViewElement;
   Element _renderViewElement;
 
+  bool _rootWidgetAttached = false;
```

Comment:
We already have https://master-api.flutter.dev/flutter/widgets/WidgetsBinding/isRootWidgetAttached.html, what's the difference between isRootWidgetAttached and _rootWidgetAttached?

Reply:
I see, the name is confusing. I cannot use isRootWidgetAttached because it schedules the first frame during attaching the root widget, the frames should be enabled before that.

Reply:
I actually have some question about https://github.com/flutter/flutter/pull/40105. What is the issue behind this pr?

Reply:
`isRootWidgetAttached` is just a getter. How does it schedule a frame?

> I actually have some question about #40105. What is the issue behind this pr?

The issue is described in the PR's description. What's your question?

Reply:
>After the change in #39535 we were only scheduling the first frame on accident as a side-effect of the warm-up frame (SchedulerBinding. scheduleWarmUpFrame awaits SchedulerBinding.endOfFrame, which implicitly schedules a frame)

I know it is a fact, but what problem will it cause if we keep this behavior?

Reply:
Everything will continue to work as before because the warmup frame schedules the first frame as a side effect. However, if we ever were to remove the warmup frame, things would start to break. Therefore, we decided to make scheduling the first frame more explicit.